theano/tensor/nnet/nnet.py

```diff
@@ -1784,6 +1803,59 @@ def graph_merge_softmax_with_crossentropy_softmax(node):
                     return [mergeable_client[1]]
 
 
+@opt.register_specialize
+@opt.register_stabilize
+@opt.register_canonicalize
+@gof.local_optimizer([CrossentropySoftmax1HotWithBiasDx])
+def local_useless_crossentropy_softmax_1hot_with_bias_dx_alloc(node):
+    """
+    Replaces a CrossentropySoftmax1HotWithBiasDx op, whose incoming gradient is
```

Comment:
I this opt tested by the change you did? If we disable this opt, do some tests fail?